server/task_check_scheduler.go

```diff
@@ -177,8 +177,77 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 	s.executors[taskType] = executor
 }
 
+// shouldScheduleTimingTaskCheck will return whether should we schedule a timing task check for current task
+// would return true when:
```

Comment:
Need to be explicit about if we need to meet any of the following conditions or all of the conditions.

The concurrent structure is missing because IIUC, the logic is 1 && ( 2 || 3)